pkg/image/registry/image/strategy.go

```diff
@@ -95,6 +95,32 @@ func (imageStrategy) ValidateUpdate(ctx kapi.Context, obj, old runtime.Object) f
 	return validation.ValidateImageUpdate(old.(*api.Image), obj.(*api.Image))
 }
 
+// signatureStrategy implements behaviour for signatures subset of image.
+type signatureStrategy struct {
+	imageStrategy
+}
+
+// SignatureStrategy is the default logic that applies when updating
+// image' signatures via the REST API.
+var SignatureStrategy = signatureStrategy{Strategy}
+
+// Canonicalize normalizes the object after validation.
+func (signatureStrategy) Canonicalize(obj runtime.Object) {
+}
+
+// PrepareForUpdate clears fields that are not allowed to be set by end users on update.
+// It allows to update only signatures field of the image.
+func (signatureStrategy) PrepareForUpdate(obj, old runtime.Object) {
+	oldImage := old.(*api.Image)
+	image := obj.(*api.Image)
+
+	image.DockerImageReference = oldImage.DockerImageReference
+	image.DockerImageMetadata = oldImage.DockerImageMetadata
+	image.DockerImageMetadataVersion = oldImage.DockerImageMetadataVersion
+	image.DockerImageManifest = oldImage.DockerImageManifest
+	image.DockerImageLayers = oldImage.DockerImageLayers
```

Comment:
@liggitt do we need to preserve all the ObjectMeta fields? I haven't found a strategy that would attempt to preserve them among upstream types.

I would allow for annotations changes. I can imagine setting annotation like `imageNotTrusted=true` in the future during signature updates.